src/Illuminate/Routing/Router.php

```diff
@@ -674,6 +677,21 @@ public static function prepareResponse($request, $response)
     }
 
     /**
+     * Create an illuminate response from a given base response object.
+     *
+     * @param  mixed  $response
+     * @return \Illuminate\Http\Response
+     */
+    private static function createResponseFromBase($response)
+    {
+        return (new Response(
+            $response->getContent(),
```

Comment:
@themsaid, with `return response()->json($model);` in my controller, I expect to be able to use `$model = $response->getOriginalContent();` in a middleware (to automatically set a `Last-Modified` header for instance).